packages/lodestar/src/chain/clock/LocalClock.ts

```diff
@@ -47,12 +46,29 @@ export class LocalClock implements IBeaconClock {
     return computeEpochAtSlot(this.config, this.currentSlot);
   }
 
-  public abort = (): void => {
-    clearTimeout(this.timeoutId);
-    if (this.signal) {
-      this.signal.removeEventListener("abort", this.abort);
+  public async waitForSlot(slot: Slot): Promise<void> {
```

Comment:
might make sense to add signal as param as well but whoever needs it can do it :)